ert_shared/ensemble_evaluator/prefect_ensemble/prefect_ensemble.py

```diff
@@ -270,30 +270,41 @@ def _fetch_input_files(self):
         return input_files
 
     def _evaluate(self, dispatch_url, ee_id):
-        sys.stderr = open("prefect.log", "w")
+        try:
+            sys.stderr = open("prefect.log", "w")
 
-        input_files = self._fetch_input_files()
+            input_files = self._fetch_input_files()
 
-        with Client(dispatch_url) as c:
-            event = CloudEvent(
-                {
-                    "type": ids.EVTYPE_ENSEMBLE_STARTED,
-                    "source": f"/ert/ee/{self._ee_id}",
-                    "datacontenttype": "application/json",
-                },
-            )
-            c.send(to_json(event).decode())
-        self.run_flow(ee_id, dispatch_url, input_files)
+            with Client(dispatch_url) as c:
+                event = CloudEvent(
+                    {
+                        "type": ids.EVTYPE_ENSEMBLE_STARTED,
+                        "source": f"/ert/ee/{self._ee_id}",
+                        "datacontenttype": "application/json",
+                    },
+                )
+                c.send(to_json(event).decode())
+            self.run_flow(ee_id, dispatch_url, input_files)
 
-        with Client(dispatch_url) as c:
-            event = CloudEvent(
-                {
-                    "type": ids.EVTYPE_ENSEMBLE_STOPPED,
-                    "source": f"/ert/ee/{self._ee_id}",
-                    "datacontenttype": "application/json",
-                },
-            )
-            c.send(to_json(event).decode())
+            with Client(dispatch_url) as c:
+                event = CloudEvent(
+                    {
+                        "type": ids.EVTYPE_ENSEMBLE_STOPPED,
+                        "source": f"/ert/ee/{self._ee_id}",
+                        "datacontenttype": "application/json",
+                    },
+                )
+                c.send(to_json(event).decode())
+        except Exception:
+            with Client(dispatch_url) as c:
+                event = CloudEvent(
+                    {
+                        "type": ids.EVTYPE_ENSEMBLE_FAILED,
+                        "source": f"/ert/ee/{self._ee_id}",
+                        "datacontenttype": "application/json",
```

Comment:
Where's the data to be sent here?

`                         "datacontenttype": "application/json", ` implies `json` will be a part of the message.

Reply:
At this point I guess we still have to think how the error  handling is supposed to work, but the data could comprise the exception that is caught.

Reply:
For distributed computing in the cloud, centralized logging of errors or metrics on errors (not very applicable for our low activity volume) are the ways to go. The former is more relevant for us.

Websockets exceptions in handlers are already logged.

Reply:
That would be one solution, just log and not send anything back, only the information that we failed, maybe that is sufficient.